pkg/cluster/deploybaseresources.go

```diff
@@ -35,6 +38,48 @@ func (m *manager) createDNS(ctx context.Context) error {
 	return m.dns.Create(ctx, m.doc.OpenShiftCluster)
 }
 
+func (m *manager) createOIDC(ctx context.Context) error {
+	if m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile != nil || m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile == nil {
+		return nil
+	}
+
+	// OIDC Storage Web Endpoint need to be determined for Development environments
+	var oidcEndpoint string
+	if m.env.FeatureIsSet(env.FeatureRequireOIDCStorageWebEndpoint) {
```

Comment:
This is needed for the Full and Local RP as the generated web endpoint is a DNS Zone present in it which cannot be determined without getting from the Storage Account Properties.
Please provide any other approach that can be taken to provide the Web Endpoint URL.

Thought around fitting it as a secret, it can be done in local dev. But for Full RP, the storage account creation depends on the Resource Group creation that is, RG Creation -> Storage Account Creation -> Fetch Web Endpoint -> Pass the Value to RP as environment variable(Can't think of a good way to do this for this flow.).

Reply:
I don't see any problem with the current approach. I agree that ideally we would store these types of configuration values in env variables, but since this DNS name is "dynamically" determined, I see why we can't do that here.